pkg/cmd/update-tls-artifacts/generate-owners/tlsmetadatainterfaces/helpers.go

```diff
@@ -23,36 +23,137 @@ func AnnotationValue(whitelistedAnnotations []certgraphapi.AnnotationValue, key
+			existing, ok := inClusterCertKeyPairs[currCert.SecretLocation]
 			if ok && !reflect.DeepEqual(existing, currCert.CertKeyInfo) {
 				errs = append(errs, fmt.Errorf("mismatch of certificate info for --namespace=%v secret/%v", currCert.SecretLocation.Namespace, currCert.SecretLocation.Name))
 				continue
 			}
 
-			certKeyPairs[currCert.SecretLocation] = currCert.CertKeyInfo
+			inClusterCertKeyPairs[currCert.SecretLocation] = currCert.CertKeyInfo
 		}
 		for i := range currPKI.InClusterResourceData.CertificateAuthorityBundles {
 			currCert := currPKI.InClusterResourceData.CertificateAuthorityBundles[i]
-			existing, ok := caBundles[currCert.ConfigMapLocation]
+			existing, ok := inClusterCABundles[currCert.ConfigMapLocation]
 			if ok && !reflect.DeepEqual(existing, currCert.CABundleInfo) {
 				errs = append(errs, fmt.Errorf("mismatch of certificate info for --namespace=%v configmap/%v", currCert.ConfigMapLocation.Namespace, currCert.ConfigMapLocation.Name))
 				continue
 			}
 
-			caBundles[currCert.ConfigMapLocation] = currCert.CABundleInfo
+			inClusterCABundles[currCert.ConfigMapLocation] = currCert.CABundleInfo
 		}
+
+		for _, tlsMetadata := range currPKI.OnDiskResourceData.TLSArtifact {
+			found := false
+			isAlsoInCluster := false
+			for _, certKeyPair := range currPKI.CertKeyPairs.Items {
+				for _, loc := range certKeyPair.Spec.OnDiskLocations {
+					if tlsMetadata.OnDiskLocation.Path == loc.Cert.Path || tlsMetadata.OnDiskLocation.Path == loc.Key.Path {
+						found = true
+						isAlsoInCluster = len(certKeyPair.Spec.SecretLocations) > 0
+						break
+					}
+				}
+				if found {
+					break
+				}
+			}
+			if found {
+				if !isAlsoInCluster {
+					info, err := GetCertKeyPairInfoForOnDiskPath(tlsMetadata.OnDiskLocation)
+					if err == nil {
+						onDiskCertsWithoutSecrets[tlsMetadata.OnDiskLocation] = info
+					}
+				}
+				continue
+			}
+
+			for _, caBundle := range currPKI.CertificateAuthorityBundles.Items {
+				for _, loc := range caBundle.Spec.OnDiskLocations {
+					if tlsMetadata.OnDiskLocation.Path == loc.Path {
+						found = true
+						isAlsoInCluster = len(caBundle.Spec.ConfigMapLocations) > 0
+						break
+					}
+				}
+				if found {
+					break
+				}
+			}
+			if found {
+				if !isAlsoInCluster {
+					info, err := GetCertificateAuthorityInfoForOnDiskPath(tlsMetadata.OnDiskLocation)
+					if err == nil {
+						onDiskCABundlesWithoutConfigMaps[tlsMetadata.OnDiskLocation] = info
+					}
+				}
+				continue
+			}
+
+		}
+
 	}
 	if len(errs) > 0 {
 		return nil, utilerrors.NewAggregate(errs)
 	}
 
-	return certs.CertsToRegistryInfo(certKeyPairs, caBundles), nil
+	return certs.CertsToRegistryInfo(inClusterCertKeyPairs, onDiskCertsWithoutSecrets, inClusterCABundles, onDiskCABundlesWithoutConfigMaps), nil
+}
+
+// TODO[vrutkovs]: move this to /api?
+
+func CertInfoForCertKeyPair(curr certgraphapi.PKIRegistryCertKeyPair) (certgraphapi.PKIRegistryCertKeyPairInfo, error) {
```

Comment:
for printing purposes. these are very different and incomparable values.  Notice how the division is kept through to the printing layer here https://github.com/openshift/origin/pull/28539/files#diff-b2792dc75c2beb82a9699af3c941303e12704132ebf6a6f102ccc7069a408a7dR178 to avoid losing the information early.